src/app/main.rs

```diff
@@ -280,6 +280,11 @@ where
         // This channel is used to move the task.
         let (resolver_bg_tx, resolver_bg_rx) = futures::sync::oneshot::channel();
 
+        let proxy_id = match config.tls_settings {
+            Conditional::Some(ref settings) => settings.pod_identity.as_ref().to_string(),
+            Conditional::None(_) => String::new(),
+        };
```

Comment:
I think the main question i have for this PR is whether the `proxy_id` should be explicitly tie dto the TLS settings or whether it should be provided as its own configuration, independently of the tls settings (though, both values would be the same).

The downside of the current approach is that an ID will only be set when there are TLS settings for the proxy. We expect this to become the default behavior in the ~2.3 time frame, but we may want a way to provide this configuration when TLS is disabled?

Reply:
My plan for this was to make a controller change where LINKERD2_PROXY_TLS_POD_IDENTITY would always be set, even when TLS is disabled.

Reply:
I don't think that this will work as you hope. `tls_settings` is only set to Some when _all_ of the TLS settings have been properly configured:

https://github.com/linkerd/linkerd2-proxy/blob/3832b49a65b9feda168cf4b96f2ed541d6fc2434/src/app/config.rs#L379-L439

To make this work, you should modify the Config type to have a `proxy_id` field which is set despite the state of the other TLS configuration.

Reply:
ack, ok, thanks